scio-test/src/test/scala/com/spotify/scio/coders/coders.scala

```diff
@@ -455,4 +455,17 @@ class CodersTest extends FlatSpec with Matchers {
     val wrappedArray: mut.WrappedArray[String] = Array("foo", "bar")
     wrappedArray coderShould notFallback()
   }
+
+  it should "support derivation of recursive types" in {
+    case class SampleField(name: String, fieldType: SampleFieldType)
+    sealed trait SampleFieldType
+    case object IntegerType extends SampleFieldType
+    case object StringType extends SampleFieldType
+    case class RecordType(fields: Vector[SampleField]) extends SampleFieldType
```

Comment:
in this test replacing `Vector` with `List` and getting as well a Coder[RecordType] should tigger the issue

Reply:
It does. I'm working on that example and trying to come up with a fix.